source/MRMesh/MRObjectLoad.cpp

```diff
@@ -44,10 +48,49 @@ std::optional<MR::IOFilter> findFilter( const MR::IOFilters& filters, const std:
         return std::nullopt;
 }
 
-} // namespace
+// Detect if mesh has enough sharp edges (>25 degrees):
+// sum length of all sharp edges greater than 1.0 * (diagonal of the bounding box)
+bool detectSharpEdges( const Mesh& mesh )
+{
+    std::atomic<double> lenSum = 0.0;
```

Comment:
we have functoin in mesh `findCreaseEdges` and then you can simply iterate over bitset to find summary length

Reply:
And constantly writing in one variable from many threads is very slow. In addition the result in floating-point will change every time. There is `tbb::parallel_deterministic_reduce` for that

Reply:
The idea of optimization is to stop early on all threads once the critical sum has been reached. Worth it?

Reply:
I think no. `lenSum.fetch_add( mesh.edgeLength( ue ) )` is terribly slow if executed for every sharp edge.